src/inc/internal/ObjectBase.hpp

```diff
@@ -19,7 +21,10 @@ namespace MSIX { namespace Meta {
 // there is exactly one value that this field is allowed to be
 template <typename T>
 static void ExactValueValidation(T value, T spec) {
-    ThrowErrorIfNot(Error::InvalidParameter, spec == value, "Incorrect value specified at field.");
+    if (spec != value) {
+        std::cout << "Value in field doesn't match expection; expected " << spec << ", got " << value << std::endl;

```

Comment:
Rather than direct output to `std::cout`, I would think it better to construct the string and include it in the thrown exception.